source/common/config/http_subscription_impl.h

```diff
@@ -75,24 +81,28 @@ class HttpSubscriptionImpl : public Http::RestApiFetcher,
     const auto status =
         google::protobuf::util::JsonStringToMessage(response.bodyAsString(), &message);
     if (status != google::protobuf::util::Status::OK) {
-      // TODO(htuch): Track stats and log failures.
+      ENVOY_LOG(warn, "REST config JSON conversion error: {}", status.ToString());
+      stats_.update_failure_.inc();
       callbacks_->onConfigUpdateFailed(nullptr);
       return;
     }
     const auto typed_resources = Config::Utility::getTypedResources<ResourceType>(message);
     try {
       callbacks_->onConfigUpdate(typed_resources);
       request_.set_version_info(message.version_info());
+      stats_.update_success_.inc();
     } catch (const EnvoyException& e) {
-      // TODO(htuch): Track stats and log failures.
+      ENVOY_LOG(warn, "REST config update rejected: {}", e.what());
+      stats_.update_rejected_.inc();
       callbacks_->onConfigUpdateFailed(&e);
     }
   }
 
   void onFetchComplete() override {}
 
   void onFetchFailure(const EnvoyException* e) override {
-    // TODO(htuch): Track stats and log failures.
+    ENVOY_LOG(warn, "REST config update failed: {}", e != nullptr ? e->what() : "");
```

Comment:
This message will look awkward in the case where e is nullptr.  Is there a generic message we can use instead of the empty string, say, "Config fetch failed" ?